### ISSUE TITLE
downloadSSURGO() HTTP error 400 

### ISSUE THREAD

**Issue by awykhuis** [ncss-tech/soilDB#286]
When using the downloadSSURGO function with a shapefile that crosses state lines in the WHERE argument, the ZIP files download for the first state, but when it try's to download ZIP files for the second state, it only partially downloads the first ZIP before receiving the following error from the curl_download function: "HTTP error 400.", any remaining ZIP files to be downloaded also receive the same error.

**Comment by brownag** [comment]
This does not seem to be a general problem with all polygon input spanning state boundaries.
For instance, this works for me on a rather slow connection:

```r
library(soilDB)

# bounding box polygon overlapping three SSAs in three states (MA, NY, VT)
x <- terra::vect("POLYGON ((-73.5956 42.6547,-73.5956 42.8423,-73.0625 42.8423,-73.0625 42.6547,-73.5956 42.6547))",
                 crs = "OGC:CRS84")

downloadSSURGO(x) 
#>  [100%] Downloaded 20139670 bytes...
#>  [100%] Downloaded 20837568 bytes...
#>  [100%] Downloaded 27590380 bytes...
#> Extracting downloaded ZIP files...
```

Can you please provide the code you are using, and full error output? and, also, the shapefile itself, or show the output of `SDA_spatialQuery()` called on your shapefile? The state boundaries shouldn't matter, what may matter is many soil survey areas you are downloading.

i.e.

```r
library(soilDB)
library(terra)
x <- vect("path/to/your/shapefile.shp")
SDA_spatialQuery(x)
```

**Comment by awykhuis** [comment]
Yes I'm seeing that now, I tried separating the shapefile by state and downloading separately, but received the same errors. I've had this issue know on several different shapefiles all spanning state lines, which is why assumed that was the issue. The code used is: 
`  downloadSSURGO(WHERE = plots,
                 destdir = soilDir,
                 extract = T,
                 include_template = T,
                 overwrite = T)`

And the results of SDA_spatialQuery() is attached.
[SDA_spatialQuery(plots).txt](https://github.com/ncss-tech/soilDB/files/11355309/SDA_spatialQuery.plots.txt)

**Comment by brownag** [comment]
OK, so I can determine from that mukey output that the 4 SSAs you are dealing with are: `c("NH607", "VT005", "VT009", "VT019")`. Sorry, what I meant to ask for was `SDA_spatialQuery(..., what="areasymbol")`, but I was able to determine areasymbols from MUKEY list.

The `downloadSSURGO()` code has not changed since last CRAN release (2.7.7). 

It seems that I can download those 4 SSAs using the areasymbols derived from mukeys, which is essentially the same as happens with shapefile input passed through `SDA_spatialQuery()`

```r
downloadSSURGO(areasymbols = c("NH607", "VT005", "VT009", "VT019"))
#> single result set, returning a data.frame
#>  [100%] Downloaded 50393764 bytes...
#>  [100%] Downloaded 30512428 bytes...
#>  [100%] Downloaded 22893380 bytes...
#>  [100%] Downloaded 23505652 bytes...
#> Extracting downloaded ZIP files...
```

The HTTP error 400 doesnt suggest an issue like this, but just to confirm: you do have write permissions in the path specified by `destdir`, correct? Also you do not need `vect()` on the destination directory, that is only needed to create a terra SpatVector object from the shapefile

**Comment by brownag** [comment]
Thanks for pointing the `include_template` bit out! Sorry I missed that, can confirm it is the problem.

Should be fixed with 74f36405, now I get:
```r
> library(soilDB)
> downloadSSURGO(areasymbols = c("NH607", "VT005", "VT009", "VT019"), 
+                include_template = T, 
+                overwrite = T)
 [100%] Downloaded 52576006 bytes...
 [100%] Downloaded 33126466 bytes...
 [100%] Downloaded 25507419 bytes...
 [100%] Downloaded 26119689 bytes...
Extracting downloaded ZIP files...
```
Basically, the issue here was Vermont has a state-specific MS Access template database. I apparently never properly tested this situation where templates are being included, with some SSA using the "US" template and some using a state template. Admittedly, I usually leave the templates out for slightly smaller download sizes. 

Basically the vectorization was broken such that the first template code (in this case "US" for New Hampshire) was used for all URLs. There is no "US" template for Vermont, so the download failed when it got to them.

**Comment by awykhuis** [comment]
Thank you for your help, seems to be working now!
